coral-erm/coral
# Issue #302: Sequenced sql files

# Information
Fixes #302. db_tools.php script now uses information about the installation/upgrade version to process any sql files that are > the current version and <= the target version.

__Example 1__: Moving from 2.1.0 => 2.4.0, the db_tools script would sequentially process any files within:
`$MODULE_NAME/install/protected/2.2.0/*.sql`
`$MODULE_NAME/install/protected/2.3.0/*.sql`
`$MODULE_NAME/install/protected/2.4.0/*.sql`

__Example 2__: On a fresh install, it will process all version files. If Coral was up to 2.4, the install script would run the following files:
`$MODULE_NAME/install/protected/test_create.sql`
`$MODULE_NAME/install/protected/install.sql`
`$MODULE_NAME/install/protected/2.0.0/*.sql`
`$MODULE_NAME/install/protected/2.1.0/*.sql`
`$MODULE_NAME/install/protected/2.2.0/*.sql`
`$MODULE_NAME/install/protected/2.3.0/*.sql`
`$MODULE_NAME/install/protected/2.4.0/*.sql`

# Workflow Moving Forward
(Based on information from #302)
1. All DB updates should be placed in the `$MODULE_NAME/install/protected/$VERSION/`, with a sequence number and issue number `$SEQ_NUM-$ISS_NUM`
2. *****__DO NOT__***** update the `install.sql` files anymore. Those represents a < 2.0 installation. All other updates should be processed sequentially in there own files.

# Testing

## Test an upgrade
1. Install a fresh copy of Coral from the master branch (or use your existing one).
2. Switch to this branch and sync code
3. nav to `your.coral.url/`
4. Make sure database is updated with any sql files from `$MODULE_NAME/install/protected/2.1.0`

## Test a fresh install
1. Which this code deployed to a server, run an install
2. Make sure database is updated with any sql files from `$MODULE_NAME/install/protected/2.1.0`
